stdlib/typing.pyi

```diff
@@ -587,69 +587,71 @@ class TextIO(IO[str]):
 class ByteString(Sequence[int], metaclass=ABCMeta): ...
 
 @_final
-class Match(Generic[AnyStr]):
+class Match(Generic[StrOrBuffer]):
     pos: int
     endpos: int
     lastindex: int | None
     lastgroup: str | None
-    string: AnyStr
+    string: StrOrBuffer
 
     # The regular expression object whose match() or search() method produced
     # this match instance.
-    re: Pattern[AnyStr]
-    def expand(self, template: AnyStr) -> AnyStr: ...
-    # group() returns "AnyStr" or "AnyStr | None", depending on the pattern.
+    re: Pattern[StrOrBuffer]
+    def expand(self, template: StrOrBuffer) -> StrOrBuffer: ...
+    # group() returns "StrOrBuffer" or "StrOrBuffer | None", depending on the pattern.
     @overload
-    def group(self, __group: _Literal[0] = ...) -> AnyStr: ...
+    def group(self, __group: _Literal[0] = ...) -> StrOrBuffer: ...
     @overload
-    def group(self, __group: str | int) -> AnyStr | Any: ...
+    def group(self, __group: str | int) -> StrOrBuffer | Any: ...
```

Comment:
This isn't right: if you match on a memoryview, you get bytes objects from .group(), not memoryviews.